packages/components/src/DeliveryReclamation.purs

```diff
@@ -0,0 +1,176 @@
+import Data.String.Read (read)
+import DatePicker.Component as DatePicker
+import Effect (Effect)
+import Effect.Aff (Aff)
+import Effect.Aff as Aff
+import Effect.Class (liftEffect)
+import Effect.Class.Console as Console
+import Effect.Now as Now
+import KSF.Grid as Grid
+import KSF.InputField.Component as InputField
+import KSF.User as User
+import Persona as Persona
+import React.Basic (JSX, make)
+import React.Basic as React
+import React.Basic.DOM as DOM
+import React.Basic.DOM.Events (preventDefault)
+import React.Basic.Events (handler, handler_)
+
+type State =
+  { publicationDate    :: Maybe DateTime
+  , claim              :: Maybe User.DeliveryReclamationClaim
+  , maxPublicationDate :: Maybe DateTime
+  , validationError    :: Maybe String
+  }
+
+type Self = React.Self Props State
+
+type Props =
+  { subsno    :: Int
+  , userUuid  :: Persona.UUID
+  , onCancel  :: Effect Unit
+  , onLoading :: Effect Unit
+  , onSuccess :: Persona.DeliveryReclamation -> Effect Unit
+  , onError   :: Persona.InvalidDateInput -> Effect Unit
+  }
+
+deliveryReclamation :: Props -> JSX
+deliveryReclamation = make component { initialState, render, didMount }
+
+initialState :: State
+initialState =
+  { publicationDate: Nothing
+  , claim: Nothing
+  , maxPublicationDate: Nothing
+  , validationError: Nothing
+  }
+
+component :: React.Component Props
+component = React.createComponent "DeliveryReclamation"
+
+didMount :: Self -> Effect Unit
+didMount self = do
+  now <- Now.nowDateTime
+  self.setState _ { maxPublicationDate = Just now }
+
+render :: Self -> JSX
+render self@{ state: { publicationDate, claim, maxPublicationDate }} =
+  DOM.div
+    { className: "clearfix delivery-reclamation--container"
+    , children:
+        [ Grid.row_
+           [ DOM.div
+               { className: "col col-11"
+               , children: [ DOM.h3_ [ DOM.text "Reklamation" ] ]
+               }
+           , DOM.div
+               { className: "col-1 flex delivery-reclamation--close-icon"
+               , children: [ DOM.div { className: "close-icon" } ]
+               , onClick: handler_ self.props.onCancel
+               }
+           , deliveryReclamationForm
+           ]
+        ]
+    }
+  where
+    deliveryReclamationForm =
+      DOM.form
+          { onSubmit: handler preventDefault (\_ -> submitForm publicationDate claim)
+          , children:
+              [ publicationDayInput
+              , claimExtensionInput
+              , claimNewDeliveryInput
+              ] `snoc`foldMap errorMessage self.state.validationError
+                `snoc` DOM.div
+                        { children: [ submitFormButton ]
+                        , className: "mt2 clearfix"
+                        }
+          }
+
+    publicationDayInput = dateInput self self.state.publicationDate "Utgivningsdatum"
+
+    claimExtensionInput =
+      InputField.inputField
+        { type_: InputField.Radio
+        , placeholder: "Placeholder"
```

Comment:
The `placeholder` prop kinda doesn't make sense when it's a radio button (maybe we should rethink `InputField` at some point), however let's have something meaningful here like `"extension"`.